src/lib/configs/project/provider.ts

```diff
@@ -46,20 +49,165 @@ export type Offer = {
+    },
+    websocketPort: {
+      nullable: true,
+      type: "number",
+      description: `Both host and container WebSocket port to use. Default: for each nox a unique port is assigned starting from ${WEB_SOCKET_PORT_START}`,
+    },
+    httpPort: {
+      nullable: true,
+      type: "number",
+      description: `Both host and container HTTP port to use. Default: for each nox a unique port is assigned starting from ${HTTP_PORT_START}`,
+    },
+    aquavmPoolSize: {
+      nullable: true,
+      type: "number",
+      description: `Number of aquavm instances to run. Default: ${DEFAULT_AQUAVM_POOL_SIZE}`,
+    },
+    systemServices: {
+      nullable: true,
+      type: "object",
+      description:
+        "System services to run by default. aquaIpfs and decider are enabled by default",
+      additionalProperties: false,
+      properties: {
+        enable: {
+          nullable: true,
+          type: "array",
+          items: { type: "string" },
+          description: `List of system services to enable`,
+        },
+        aquaIpfs: {
+          type: "object",
+          description: "Aqua IPFS service configuration",
+          additionalProperties: false,
+          nullable: true,
+          properties: {
+            externalApiMultiaddr: {
+              nullable: true,
+              type: "string",
+              description: `Multiaddress of external IPFS API`,
+            },
+            localApiMultiaddr: {
+              nullable: true,
+              type: "string",
+              description: `Multiaddress of local IPFS API`,
+            },
+          },
+          required: [],
+        },
+        decider: {
+          type: "object",
+          description: "Decider service configuration",
+          additionalProperties: false,
+          nullable: true,
+          properties: {
+            deciderPeriodSec: {
+              nullable: true,
+              type: "number",
+              description: `Decider period in seconds`,
+            },
+            workerIpfsMultiaddr: {
+              nullable: true,
+              type: "string",
+              description: `Multiaddress of worker IPFS node`,
+            },
+            networkApiEndpoint: {
+              nullable: true,
+              type: "string",
+              description: `Network API endpoint`,
+            },
+            networkId: {
+              nullable: true,
+              type: "number",
+              description: `Network ID`,
+            },
+            startBlock: {
+              nullable: true,
+              type: "string",
+              description: `Start block`,
+            },
+            matcherAddress: {
+              nullable: true,
+              type: "string",
+              description: `Matcher address`,
+            },
+            walletKey: {
+              nullable: true,
```

Comment:
I think it should be `nullable: false`, because Decider doesn't have a default value for `walletKey`

Reply:
for now if user omits it - cli will pick one of the keys from the local network, but we should setup some way of generating these keys. We discussed using files previously. Maybe files should be used and passed to containers and this property should be removed completely from here, I just temporarely did this so tests work at least

Reply:
I think that's OK.

> cli will pick one of the keys from the local network

I understand this as fcli has pre-generated list of keys somewhere. If my understanding is correct, what will happen if a user generates more Noxes than you have pre-generated keys?

Reply:
currently it will cycle through them and use the same for multipple peers, I am thinking to maybe add a warning about that when I put the keys in the files. But probably when running local env users will rarely need more than 10 and we have 20 hardcoded keys

Reply:
Maybe it's easier to generate keys? I think you can simply use 32-byte array as a Private Key, and increment last numbers in it.

Reply:
I can ofcourse generate keys, but will it be a problem in the furture that these keys don't have any money on their account? The ones that I use have some by default
If it's always ok for compute peers to have empty accounts - then generating seems to be a better option